src/refdb_fs.c

```diff
@@ -494,6 +646,86 @@ static int packed_lookup(
 	return error;
 }
 
+static int packed_lookup(
+        git_reference **out,
+        refdb_fs_backend *backend,
+        const char *ref_name)
+{
+	int error = 0;
+	const char *left, *right, *data_end;
+
+	if ((error = packed_map_check(backend)) < 0)
+		return error;
+
+	if (!backend->sorted) {
+		return packed_unsorted_lookup(out, backend, ref_name);
+	}
+
+	left = backend->packed_refs_map.data;
+	right = data_end = ((const char *)backend->packed_refs_map.data) +
```

Comment:
This cast should be unnecessary?  🤔 
```suggestion
	right = data_end = backend->packed_refs_map.data +
```